UserDefaults CRUD actions for favourites.

steven-hill opened the pull request:
This allows the user to have a list separate from the search results. The user can easily navigate to the list via the tab bar.

It is handled by a Persistence Manager and a Favourites Coordinator.

The “Add to favourites” button title label in MovieDetailVC changes when tapped to tell the user the movie was added to the favourites list. The delete action is shown via swipe to delete on the table view cell in FavouritesListVC.

To be added - an alert to tell the user if the UserDefaults actions fail.

steven-hill committed:
I have completed the UserDefaults actions of retrieving, adding, saving and deleting favourites.

This allows the user to have a list separate from the search results. The user can easily navigate to the list via the tab bar.

It is handled by a Persistence Manager and a Favourites Coordinator.

The “Add to favourites” button title label in MovieDetailVC changes when tapped to tell the user the movie was added to the favourites list. The delete action is shown via swipe to delete on the table view cell in FavouritesListVC.

To be added - an alert to tell the user if the UserDefaults actions fail.

OlivAlex commented on MovieList/MovieList/Favourites/PersistenceManager.swift:
```diff
@@ -9,46 +9,48 @@ import Foundation
-            favourites.append(newFavourite)
-            return true
-        }) else { return }
+    func addToFavourites(name: String) {
```
Possibly rename to: `func saveMovieNameToFavourites(newMovie: String)`

steven-hill replied:
Renamed to: func saveMovieNameToFavourites(newMovie: String).

OlivAlex reviewed (commented):
LGTM!

steven-hill committed:
Renamed a method

func addToFavourites(name: String) renamed to func saveMovieNameToFavourites(newMovie: String). 

Also updated let newFavourite = name to let newFavourite = newMovie.

steven-hill committed:
Updated method name 

A method was renamed in PersistenceManager, so it needed to be renamed here also.

persistenceManager.addToFavourites(name: name) renamed to persistenceManager.saveMovieNameToFavourites(newMovie: name).

steven-hill commented:
After completing the changes, the app was tested on the simulator. All CRUD actions still work as expected.

OlivAlex committed:
Creating .gitignore file for project

OlivAlex committed:
Merge branch 'main' into merge-main-to-UserDefaults
